hawkular-alerts-api/src/main/java/org/hawkular/alerts/api/model/event/Alert.java

```diff
@@ -181,6 +184,31 @@ public void addNote(String user, String text) {
         getNotes().add(new Note(user, text));
     }
 
+    public List<LifeCycle> getLifecycle() {
+        if (null == lifecycle) {
+            this.lifecycle = new ArrayList<>();
+        }
+        return lifecycle;
+    }
+
+    public void setLifecycle(List<LifeCycle> lifecycle) {
+        this.lifecycle = lifecycle;
+    }
+
+    public void addLifecycle(Status status, String user, long stime) {
+        if (status == null || user == null) {
+            throw new IllegalArgumentException("Lifecycle must have non-null state and user");
+        }
+        getLifecycle().add(new LifeCycle(status, user, stime));
+    }
+
+    public LifeCycle lastLifecycle() {
```

Comment:
Should this have a "get" prefix?  Also, would "Current" maybe be better than "Last"?